dot/rpc/modules/state.go

```diff
@@ -131,6 +125,32 @@ func NewStateModule(net NetworkAPI, storage StorageAPI, core CoreAPI) *StateModu
 	}
 }
 
+// GetPairs returns the keys with prefix, leave empty to get all the keys.
+func (sm *StateModule) GetPairs(r *http.Request, req *[]string, res *[]interface{}) error {
+	// TODO implement handling of block hash parameter (See issue #834)
+	pReq := *req
+	reqBytes, _ := common.HexToBytes(pReq[0])
+
+	if len(reqBytes) < 1 {
+		pairs := sm.storageAPI.Entries()
+		for k, v := range pairs {
+			*res = append(*res, []string{"0x" + hex.EncodeToString([]byte(k)), "0x" + hex.EncodeToString(v)})
+		}
+	} else {
+		resI, err := sm.storageAPI.GetStorage(reqBytes)
+		if err != nil {
+			return err
+		}
+		if resI != nil {
+			*res = append(*res, []string{"0x" + hex.EncodeToString(reqBytes), "0x" + hex.EncodeToString(resI)})
+		} else {
+			*res = []interface{}{}
+		}
```

Comment:
I think this only gets one value from a key, but it should get all keys with the the same prefix? this functionality will need to be implemented into the trie since it doesn't exist yet, I can make an issue for it

Reply:
could you add a TODO for #837 somewhere?

Reply:
Done.

Reply:
Done.